src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,16 +48,36 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                string[] symbols = conditionMemberName.Split('.');
                 Type declaringType = testMethodDeclaringType;
 
-                if (symbols.Length == 2)
+                // We have qualified type name with the assembly name, something like
+                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
```

Comment:
the whole string passed as the attribute parameter would be 

```
System.PlatformDetection, CoreFx.Private.TestUtilities!IsNonZeroLowerBoundArraySupported`
```

which is correct. the part before the '!'is the type name and the part after '!' is the method name. I am not sure what is wrong?